astropy/utils/xml/iterparser.py

```diff
@@ -67,18 +67,17 @@ def _convert_to_fd_or_read_function(fd):
         if sys.platform.startswith('win'):
             yield new_fd.read
         else:
-            if six.PY3:
-                if isinstance(new_fd, io.FileIO):
+            if six.PY2:
+                if isinstance(new_fd, file):
                     yield new_fd
                 else:
                     yield new_fd.read
-            elif six.PY2:
-                if isinstance(new_fd, file):
+            else:
+                if isinstance(new_fd, io.FileIO):
                     yield new_fd
                 else:
                     yield new_fd.read
 
-
```

Comment:
This blank line should not be deleted, as it is recommended that two blank lines separate different functions.